Documenting and cleaning backup manager

Arthur-Milchior opened the pull request:
I need to consider free space to decide whether to offer to export the collection before storage migration.
Lots of the related functions are in the BackupManager. So I went, read it, clarify names which are not entirely clear, document it, and while I'm at it, improving some functions. 

In particular, some functions have default values that are absolutely not explicit; now it's explicitly documented what occurs when we don't know how much space there remains on disk

mikehardy commented on AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerBackupNoSpaceLeftDialog.kt:
```diff
@@ -28,7 +28,8 @@ class DeckPickerBackupNoSpaceLeftDialog : AnalyticsDialogFragment() {
         val res = resources
-        val space = BackupManager.getFreeDiscSpace(CollectionHelper.getCollectionPath(requireActivity()))
+        // If we show this dialog, we assume we are able to measure the free space
+        val space = BackupManager.getFreeDiscSpace(CollectionHelper.getCollectionPath(requireActivity()))!!
```
Is this `!!` safe here, after going to the trouble to make it nullable here?

Arthur-Milchior replied:
The comment tried to explain why it is. We call it only if we were able to measure the free space. So the only case where it would create an issue is we lost the ability in between the deck picker and this dialog opening.

If you prefer, I could otherwise put it in the Bundle. That would even avoid requiring to compute it twice

Arthur-Milchior commented:
Merge resolved

Arthur-Milchior commented:
I removed the !!. Instead, I just changed the message so that it claims there is not enough free space without trying to make a guess at the actual amount

david-allison commented on AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt:
```diff
@@ -1098,7 +1098,9 @@ open class DeckPicker :
             }
         }
-        if (!BackupManager.enoughDiscSpace(CollectionHelper.getCurrentAnkiDroidDirectory(this))) {
+        // If we can't measure free space, let the user uses the app.
```
`use`

david-allison commented on AnkiDroid/src/main/res/values/09-backup.xml:
```diff
@@ -20,6 +20,7 @@
 --><resources>
 
     <string name="backup_deck_no_space_left">Backup not saved. Not enough space left on SD card. Lower the backup depth or remove some other files.</string>
+    <string name="sd_space_warning_unknown">Not enough space on your SD card left.\nFree some space to avoid data loss.</string>
```
Let's not use `SD Card`, I don't believe this is accurate

Arthur-Milchior replied:
In this case, we should change it in all other related strings
